diff --git a/vesc_ackermann/src/vesc_to_odom.cpp b/vesc_ackermann/src/vesc_to_odom.cpp
index 86dc28d..488ccff 100644
--- a/vesc_ackermann/src/vesc_to_odom.cpp
+++ b/vesc_ackermann/src/vesc_to_odom.cpp
@@ -98,6 +98,7 @@ void VescToOdom::vescStateCallback(const VescStateStamped::SharedPtr state)

   // convert to engineering units
   double current_speed = (-state->state.speed - speed_to_erpm_offset_) / speed_to_erpm_gain_;
+  current_speed *= -1;
   if (std::fabs(current_speed) < 0.05) {
     current_speed = 0.0;
   }
